nixos/modules/services/security/vaultwarden/default.nix

```diff
@@ -235,6 +236,10 @@ in {
       };
       wantedBy = [ "multi-user.target" ];
     };
+
+    systemd.tmpfiles.rules = optional (cfg.backupDir != null) ''
```

Comment:
Shouldn't this use mkIf? I am not sure

Reply:
I can see that both `lib.optional` and `lib.mkIf` is used throughout nixpkgs. Not sure which one would be preferred though.

https://github.com/NixOS/nixpkgs/blob/f320b0d46ee33f521f50d7c6885605714210e5c9/nixos/modules/virtualisation/qemu-vm.nix#L1009

https://github.com/NixOS/nixpkgs/blob/f320b0d46ee33f521f50d7c6885605714210e5c9/nixos/modules/system/boot/tmp.nix#L67

Reply:
I think the only difference is that lib.optional returns an empty array on false which is sometimes not desired because it creates the attr path.